app/javascript/packs/admin/editUser.jsx

```diff
@@ -1,4 +1,44 @@
+import { h, render } from 'preact';
+import { TagAutocompleteSelection } from '../../article-form/components/TagAutocompleteSelection';
+import { TagAutocompleteOption } from '../../article-form/components/TagAutocompleteOption';
+import { MultiSelectAutocomplete } from '@crayons';
 import { initializeDropdown } from '@utilities/dropdownUtils';
+import { fetchTags } from '@utilities/search';
+import { getCsrfToken } from '@utilities/getUserDataAndCsrfToken';
+
+window.getCsrfToken = getCsrfToken;
+
+const tagsRoot = document.getElementById('tag-moderation-container');
+function initializeTagModeration() {
+  if (!tagsRoot) {
+    return;
+  }
+
+  const form = tagsRoot.closest('form');
+  const currentModTags = JSON.parse(tagsRoot.dataset.currentModTags);
+  const modTags = form['user[mod_tags]'];
+
+  // Converts the array of selected items into a plain string to be saved in the article form
+  const syncSelections = (selections = []) => {
+    modTags.value = JSON.stringify(selections.map(({ id }) => id));
+  };
+
+  render(
+    <MultiSelectAutocomplete
+      defaultValue={currentModTags}
+      fetchSuggestions={fetchTags}
+      labelText="Assign tags"
+      placeholder="Add a tag..."
+      showLabel={false}
+      SuggestionTemplate={TagAutocompleteOption}
+      SelectionTemplate={TagAutocompleteSelection}
+      onSelectionsChanged={syncSelections}
+      // onlySuggestions={true}
```

Comment:
TODO: Sort this out. This was kind of working, but it breaks how the multi-select works. TLDR, I want to avoid adding new tags because a potential moderator can only moderate existing tags.TODO: Sort this out. This was kind of working, but it breaks how the multi-select works. TLDR, I want to avoid adding new tags because a potential moderator can only moderate existing tags.

Reply:
Just an idea, but I think perhaps the `MultiSelectAutocomplete` shouldn't be responsible for this. 

So currently:
1. We pass in `fetchSuggestions` which returns only existing matching tags from the search call
2. In `MultiSelectAutocomplete` if the results are empty, we add the current search term to the suggestions

Instead we could:
- Not do point 2 above
- In `TagsField` which we use for the editor, we could make sure the `fetchSuggestions` prop we pass in adds the current search term as a suggestion if nothing was found in the search call

Reply:
I got this working yesterday, my logic was wrong but let me try out your suggestion.

Reply:
I've got it working as per your suggestion. Given the component only adds tags return from the `fetchSuggestions` prop now,  I'm going to alter some of the component tests as they assumed non-existent tags could be added.